From d696a96249a7bf2e71977594a278a86b06e50cb6 Mon Sep 17 00:00:00 2001
From: Jonathan Dieter <jdieter@gmail.com>
Date: Sat, 22 Dec 2018 16:56:59 +0000
Subject: [PATCH] Zchunk configuration flags were backwards, so setting zchunk=false would

---
 libdnf/repo/Repo.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libdnf/repo/Repo.cpp b/libdnf/repo/Repo.cpp
index 42823e7..7c0e464 100644
--- a/libdnf/repo/Repo.cpp
+++ b/libdnf/repo/Repo.cpp
@@ -575,7 +575,7 @@ std::unique_ptr<LrHandle> Repo::Impl::lrHandleInitLocal()
     handleSetOpt(h.get(), LRO_LOCAL, 1L);
 #ifdef LRO_SUPPORTS_CACHEDIR
     /* If zchunk is enabled, set librepo cache dir */
-    if (!conf->getMasterConfig().zchunk().getValue())
+    if (conf->getMasterConfig().zchunk().getValue())
         handleSetOpt(h.get(), LRO_CACHEDIR, conf->basecachedir().getValue().c_str());
 #endif
     return h;
@@ -662,7 +662,7 @@ std::unique_ptr<LrHandle> Repo::Impl::lrHandleInitRemote(const char *destdir, bo
 
 #ifdef LRO_SUPPORTS_CACHEDIR
     /* If zchunk is enabled, set librepo cache dir */
-    if (!conf->getMasterConfig().zchunk().getValue())
+    if (conf->getMasterConfig().zchunk().getValue())
         handleSetOpt(h.get(), LRO_CACHEDIR, conf->basecachedir().getValue().c_str());
 #endif
 
--
libgit2 0.27.7

