From 44c1108cf2d06eb454dd29d09ca95d53ee24f88a Mon Sep 17 00:00:00 2001
From: Lukáš Hrázký <lhrazky@redhat.com>
Date: Wed, 17 Apr 2019 13:48:26 +0200
Subject: [PATCH] Set relevant g_log domain handlers instead of a default one

---
 python/hawkey/sack-py.cpp | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/python/hawkey/sack-py.cpp b/python/hawkey/sack-py.cpp
index 11f9d7c..e582268 100644
--- a/python/hawkey/sack-py.cpp
+++ b/python/hawkey/sack-py.cpp
@@ -187,15 +187,27 @@ log_handler(const gchar *log_domain, GLogLevelFlags log_level, const gchar *mess
     g_free(msg);
 }
 
+static void
+log_handler_noop(const gchar *, GLogLevelFlags, const gchar *, gpointer)
+{
+}
+
 gboolean
 set_logfile(const gchar *path, FILE *log_out)
 {
     log_out = fopen(path, "a");
 
     if (!log_out)
         return FALSE;
 
-    g_log_set_default_handler(log_handler, log_out);
+    // The default log handler prints messages that weren't handled by any
+    // other logger to stderr/stdout, we do not want that
+    g_log_set_default_handler(log_handler_noop, nullptr);
+
+    // set the handler for the default domain as well as "libdnf"
+    g_log_set_handler(nullptr, G_LOG_LEVEL_MASK, log_handler, log_out);
+    g_log_set_handler("libdnf", G_LOG_LEVEL_MASK, log_handler, log_out);
+
     g_info("=== Started libdnf-%d.%d.%d ===", LIBDNF_MAJOR_VERSION,
             LIBDNF_MINOR_VERSION, LIBDNF_MICRO_VERSION);
     return TRUE;
--
libgit2 0.27.8

